sdk/storage_datalake/examples/filesystem.rs

```diff
@@ -1,20 +1,20 @@
-use azure_storage::storage_shared_key_credential::StorageSharedKeyCredential;
+use azure_storage::prelude::StorageCredentials;
 use azure_storage_datalake::prelude::*;
 use futures::stream::StreamExt;
 use std::num::NonZeroU32;
 use time::OffsetDateTime;
 
 #[tokio::main]
 async fn main() -> azure_core::Result<()> {
-    let data_lake_client = create_data_lake_client().await.unwrap();
+    let data_lake_client = create_data_lake_client();
 
     let file_system_name = format!(
         "azurerustsdk-datalake-example00-{}",
         OffsetDateTime::now_utc().unix_timestamp()
     );
     let file_system_client = data_lake_client
         .clone()
```

Comment:
Can we have the sub-client factory methods call `self.clone()`? That is the pattern in the generated clients:
https://github.com/Azure/azure-sdk-for-rust/blob/f99f9c1944a25b34420424daa792bb21c37ba6ba/services/svc/blobstorage/src/package_2021_08/mod.rs#L106-L108

Reply:
They already do this. The `clone` here is redundant. I will delete it.